include/bcs/FieldEmissionBC.h

```diff
@@ -22,24 +22,24 @@ class FieldEmissionBC : public ADIntegratedBC
 protected:
   virtual ADReal computeQpResidual() override;
 
-  Real _r_units;
-  Real _r;
+  const Real _r_units;
+  const Real _r;
 
+  const unsigned int _num_ions;
   // Coupled variables
 
   const ADVariableGradient & _grad_potential;
-  const ADVariableValue & _mean_en;
-  MooseVariable & _ip_var;
-  const ADVariableValue & _ip;
-  const ADVariableGradient & _grad_ip;
+  std::vector<MooseVariable *> _ip_var;
+  std::vector<const ADVariableValue *> _ip;
+  std::vector<const ADVariableGradient *> _grad_ip;
 
   const ADMaterialProperty<Real> & _muem;
   const MaterialProperty<Real> & _massem;
   const MaterialProperty<Real> & _e;
-  const MaterialProperty<Real> & _sgnip;
-  const ADMaterialProperty<Real> & _muip;
-  const ADMaterialProperty<Real> & _Dip;
-  const MaterialProperty<Real> & _se_coeff;
+  std::vector<const MaterialProperty<Real> *> _sgnip;
+  std::vector<const ADMaterialProperty<Real> *> _muip;
+  std::vector<const ADMaterialProperty<Real> *> _Dip;
+  const std::vector<Real> _se_coeff;
```

Comment:
Since there is so much functionality within MOOSE that I do not fully understand I cannot say for sure that there is not a case where being able to change the address of these pointers so I will remove the const modified from this and similar files. Based on what I know about MOOSE I think the cases where this may be true would be quite small. Maybe @csdechant and @lindsayad might have a better answer.

Reply:
I am usually against removing flexibility. As for secondary electron coefficients, there are cases where they are energy dependent (either on the ion or material temperature). I know I introduce some BC's that has the secondary electron coefficient as a real input parameter, instead of a material property. This was because I wanted to have different constant secondary electron coefficients on different boundaries, but the secondary electron coefficient material property and electron transport coefficient material property were coded in the same material object. In hindsight, I should of just separated the secondary electron coefficient into a new material object and have these material object only applied to the different boundaries.

Reply:
I will go ahead and remove this const modifier then. I think if we want make secondary electron emission coefficients a material property then that should be opened as an issue and addressed in a separate PR.

Reply:
Somehow I missed these updates - apologies. @csdechant is correct. I was referring to it being made a Real value (a constant number) vs being a `MaterialProperty<Real>` which could be spatially- and/or time-varying or changing in relation to another variable.

The `const` modifier is fine to leave as it is. I generally like as much `const`-ness as we can get away with for safety.

Reply:
I see what you're saying now. I can make the secondary electron emission coefficients a material properties then it shouldn't be much extra work to make this change.